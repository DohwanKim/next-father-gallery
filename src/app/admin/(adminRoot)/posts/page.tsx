import Link from 'next/link';

import Posts from '@/components/admin/organism/Posts';

export default async function PostPage() {
  return (
    <>
      <h1>Admin Posts</h1>
      <Link href={'/admin/posts/new'}>등록하기</Link>
      <Posts />
    </>
  );
}
