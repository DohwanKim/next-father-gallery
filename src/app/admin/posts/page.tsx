import Posts from '@/components-admin/Posts';

import { css } from '../../../../styled-system/css';

export default function Page() {
  return (
    <>
      <h1 className={css({ fontSize: '22px', fontWeight: 'bold', mb: '16px' })}>
        Admin Posts
      </h1>
      <Posts />
    </>
  );
}
