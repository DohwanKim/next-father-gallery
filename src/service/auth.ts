import apiFetch from '@/utils/fetchIntance';

export const signIn = async (data: { username: string; password: string }) => {
  return await apiFetch('/api/auth/signin', {
    method: 'POST',
    body: data,
  });
};

export const signOut = async () => {
  return await apiFetch('/api/auth/signout', {
    method: 'POST',
  });
};

export const refreshToken = async () => {
  return await apiFetch('/api/auth/refresh').catch(() => {
    throw new Error('로그인 페이지로 이동');
  });
};
