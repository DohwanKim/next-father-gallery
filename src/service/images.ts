import apiFetch from '@/utils/fetchIntance';

type ImageUploadUrl = {
  uploadUrl: string;
};

export const getImageUploadUrl = async (): Promise<string> => {
  return await apiFetch<ImageUploadUrl>('/api/images/get-upload-url').then(
    (res) => res.body.uploadUrl,
  );
};

export const uploadImage = async (url: string, file: FileList) => {
  const formData = new FormData();
  formData.append('file', file[0]);

  return await fetch(url, {
    method: 'POST',
    body: formData,
  })
    .then((res) => res.json())
    .then((res) => res.result);
};
