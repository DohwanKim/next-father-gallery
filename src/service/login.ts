import apiFetch from '@/utils/ofetchIntance';

export const signIn = async (data: { username: string; password: string }) => {
  return await apiFetch('/auth/signin', {
    method: 'POST',
    body: data,
  });
};
