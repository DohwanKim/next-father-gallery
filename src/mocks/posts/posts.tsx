import { http, HttpResponse } from 'msw';

import { Paginate } from '@/types/paginate.type';
import { Post } from '@/types/posts.type';

const BASE_URL = `${process.env.NEXT_PUBLIC_BASE_API_URL}`;
const paginatePosts: Paginate<Post> = {
  items: [
    {
      id: 25,
      createAt: '2024-02-21T11:32:53.093Z',
      updateAt: '2024-03-05T06:16:39.649Z',
      version: 7,
      title: 'MOCK 샘플 게시글1',
      drawingDate: null,
      artType: 'WATERCOLOR',
      canvasSize: '61x45cm',
      price: 10000,
      isSold: true,
      frameType: 'Paper',
      contents: '이쁜 그림을 그려보았습니다.',
      tags: [],
      img: {
        uid: 25,
        id: 'f0a8866e-2b9a-4329-ba01-0ae7fc210b00',
        filename: 'KakaoTalk_20240105_194605293-crop-crop.jpg',
        metadata: null,
        uploaded: '2024-02-21T11:32:52.918Z',
        requireSignedURLs: false,
        variants: [
          'https://imagedelivery.net/MXlZJaCYonU_kO5E66JLvw/f0a8866e-2b9a-4329-ba01-0ae7fc210b00/public',
          'https://imagedelivery.net/MXlZJaCYonU_kO5E66JLvw/f0a8866e-2b9a-4329-ba01-0ae7fc210b00/adminPost',
          'https://imagedelivery.net/MXlZJaCYonU_kO5E66JLvw/f0a8866e-2b9a-4329-ba01-0ae7fc210b00/origin',
        ],
      },
    },
    {
      id: 30,
      createAt: '2024-03-05T08:32:08.214Z',
      updateAt: '2024-03-05T08:32:08.214Z',
      version: 1,
      title: '1',
      drawingDate: null,
      artType: 'WATERCOLOR',
      canvasSize: '',
      price: 0,
      isSold: false,
      frameType: '',
      contents: '',
      tags: [],
      img: {
        uid: 30,
        id: 'a2bc0c8c-91aa-4724-6992-a70b2e0d9d00',
        filename: 'KakaoTalk_20240220_115102262-crop (1).jpg',
        metadata: null,
        uploaded: '2024-03-05T08:32:07.708Z',
        requireSignedURLs: false,
        variants: [
          'https://imagedelivery.net/MXlZJaCYonU_kO5E66JLvw/a2bc0c8c-91aa-4724-6992-a70b2e0d9d00/origin',
          'https://imagedelivery.net/MXlZJaCYonU_kO5E66JLvw/a2bc0c8c-91aa-4724-6992-a70b2e0d9d00/adminPost',
          'https://imagedelivery.net/MXlZJaCYonU_kO5E66JLvw/a2bc0c8c-91aa-4724-6992-a70b2e0d9d00/public',
        ],
      },
    },
    {
      id: 33,
      createAt: '2024-03-05T08:32:59.030Z',
      updateAt: '2024-03-05T08:32:59.030Z',
      version: 1,
      title: '4',
      drawingDate: null,
      artType: 'WATERCOLOR',
      canvasSize: '',
      price: 0,
      isSold: false,
      frameType: '',
      contents: '',
      tags: [],
      img: {
        uid: 33,
        id: '2a3d6e6c-4448-4ebc-1628-1d21fc6bdc00',
        filename: '스크린샷 2024-01-09 오후 5.41.05.png',
        metadata: null,
        uploaded: '2024-03-05T08:32:58.806Z',
        requireSignedURLs: false,
        variants: [
          'https://imagedelivery.net/MXlZJaCYonU_kO5E66JLvw/2a3d6e6c-4448-4ebc-1628-1d21fc6bdc00/adminPost',
          'https://imagedelivery.net/MXlZJaCYonU_kO5E66JLvw/2a3d6e6c-4448-4ebc-1628-1d21fc6bdc00/origin',
          'https://imagedelivery.net/MXlZJaCYonU_kO5E66JLvw/2a3d6e6c-4448-4ebc-1628-1d21fc6bdc00/public',
        ],
      },
    },
    {
      id: 31,
      createAt: '2024-03-05T08:32:22.964Z',
      updateAt: '2024-03-05T08:32:22.964Z',
      version: 1,
      title: '2',
      drawingDate: null,
      artType: 'WATERCOLOR',
      canvasSize: '',
      price: 0,
      isSold: false,
      frameType: '',
      contents: '',
      tags: [],
      img: {
        uid: 31,
        id: '408a6035-6382-4a8e-2cba-a8be125a1300',
        filename: '94D1F7DA-46FC-467D-824C-ED7678A29011_4_5005_c.jpeg',
        metadata: null,
        uploaded: '2024-03-05T08:32:22.471Z',
        requireSignedURLs: false,
        variants: [
          'https://imagedelivery.net/MXlZJaCYonU_kO5E66JLvw/408a6035-6382-4a8e-2cba-a8be125a1300/public',
          'https://imagedelivery.net/MXlZJaCYonU_kO5E66JLvw/408a6035-6382-4a8e-2cba-a8be125a1300/adminPost',
          'https://imagedelivery.net/MXlZJaCYonU_kO5E66JLvw/408a6035-6382-4a8e-2cba-a8be125a1300/origin',
        ],
      },
    },
    {
      id: 32,
      createAt: '2024-03-05T08:32:37.003Z',
      updateAt: '2024-03-05T08:32:37.003Z',
      version: 1,
      title: '3',
      drawingDate: null,
      artType: 'WATERCOLOR',
      canvasSize: '',
      price: 0,
      isSold: false,
      frameType: '',
      contents: '',
      tags: [],
      img: {
        uid: 32,
        id: 'f293b455-af60-4b31-47ca-c1ca219a4c00',
        filename: '18d4b7abc913b0a.jpg',
        metadata: null,
        uploaded: '2024-03-05T08:32:36.907Z',
        requireSignedURLs: false,
        variants: [
          'https://imagedelivery.net/MXlZJaCYonU_kO5E66JLvw/f293b455-af60-4b31-47ca-c1ca219a4c00/public',
          'https://imagedelivery.net/MXlZJaCYonU_kO5E66JLvw/f293b455-af60-4b31-47ca-c1ca219a4c00/origin',
          'https://imagedelivery.net/MXlZJaCYonU_kO5E66JLvw/f293b455-af60-4b31-47ca-c1ca219a4c00/adminPost',
        ],
      },
    },
    {
      id: 34,
      createAt: '2024-03-05T08:33:21.070Z',
      updateAt: '2024-03-05T08:33:21.070Z',
      version: 1,
      title: '5',
      drawingDate: null,
      artType: 'WATERCOLOR',
      canvasSize: '',
      price: 0,
      isSold: false,
      frameType: '',
      contents: '',
      tags: [],
      img: {
        uid: 34,
        id: '1b4de31f-e685-4c61-0554-0873c1784f00',
        filename: 'IMG_4786.jpeg',
        metadata: null,
        uploaded: '2024-03-05T08:33:20.864Z',
        requireSignedURLs: false,
        variants: [
          'https://imagedelivery.net/MXlZJaCYonU_kO5E66JLvw/1b4de31f-e685-4c61-0554-0873c1784f00/origin',
          'https://imagedelivery.net/MXlZJaCYonU_kO5E66JLvw/1b4de31f-e685-4c61-0554-0873c1784f00/adminPost',
          'https://imagedelivery.net/MXlZJaCYonU_kO5E66JLvw/1b4de31f-e685-4c61-0554-0873c1784f00/public',
        ],
      },
    },
    {
      id: 24,
      createAt: '2024-02-21T11:11:45.876Z',
      updateAt: '2024-03-05T06:16:25.547Z',
      version: 3,
      title: '제목',
      drawingDate: null,
      artType: 'NONE',
      canvasSize: '',
      price: 0,
      isSold: false,
      frameType: '',
      contents: '',
      tags: [],
      img: {
        uid: 24,
        id: 'b6655951-1826-493f-5adf-a9ca2ebbb800',
        filename: 'colorful_background_18-wallpaper-7680x4800.jpg',
        metadata: null,
        uploaded: '2024-02-21T11:11:45.634Z',
        requireSignedURLs: false,
        variants: [
          'https://imagedelivery.net/MXlZJaCYonU_kO5E66JLvw/b6655951-1826-493f-5adf-a9ca2ebbb800/public',
          'https://imagedelivery.net/MXlZJaCYonU_kO5E66JLvw/b6655951-1826-493f-5adf-a9ca2ebbb800/adminPost',
        ],
      },
    },
    {
      id: 23,
      createAt: '2024-02-21T10:46:23.694Z',
      updateAt: '2024-03-05T06:16:32.971Z',
      version: 3,
      title: '이미지 리사이징 테스트',
      drawingDate: null,
      artType: 'WATERCOLOR',
      canvasSize: '',
      price: 0,
      isSold: false,
      frameType: '',
      contents: '',
      tags: [],
      img: {
        uid: 23,
        id: 'c89d41d6-1c47-4517-3e39-e68c7a5d0800',
        filename: '18d4b7abc913b0a.jpg',
        metadata: null,
        uploaded: '2024-02-21T10:46:23.561Z',
        requireSignedURLs: false,
        variants: [
          'https://imagedelivery.net/MXlZJaCYonU_kO5E66JLvw/c89d41d6-1c47-4517-3e39-e68c7a5d0800/adminPost',
          'https://imagedelivery.net/MXlZJaCYonU_kO5E66JLvw/c89d41d6-1c47-4517-3e39-e68c7a5d0800/public',
        ],
      },
    },
    {
      id: 35,
      createAt: '2024-03-05T08:33:39.895Z',
      updateAt: '2024-03-05T08:33:39.895Z',
      version: 1,
      title: '6',
      drawingDate: null,
      artType: 'NONE',
      canvasSize: '',
      price: 0,
      isSold: false,
      frameType: '',
      contents: '',
      tags: [],
      img: {
        uid: 35,
        id: '0f6f326b-b35c-4d0c-a952-4b6881dc3100',
        filename: '스크린샷 2024-01-10 오후 6.02.29.png',
        metadata: null,
        uploaded: '2024-03-05T08:33:39.768Z',
        requireSignedURLs: false,
        variants: [
          'https://imagedelivery.net/MXlZJaCYonU_kO5E66JLvw/0f6f326b-b35c-4d0c-a952-4b6881dc3100/origin',
          'https://imagedelivery.net/MXlZJaCYonU_kO5E66JLvw/0f6f326b-b35c-4d0c-a952-4b6881dc3100/adminPost',
          'https://imagedelivery.net/MXlZJaCYonU_kO5E66JLvw/0f6f326b-b35c-4d0c-a952-4b6881dc3100/public',
        ],
      },
    },
    {
      id: 36,
      createAt: '2024-03-05T08:33:59.901Z',
      updateAt: '2024-03-05T08:33:59.901Z',
      version: 1,
      title: '7',
      drawingDate: null,
      artType: 'WATERCOLOR',
      canvasSize: '',
      price: 0,
      isSold: false,
      frameType: '',
      contents: '',
      tags: [],
      img: {
        uid: 36,
        id: '5ccbd20a-49de-48dc-f878-75ece31e7600',
        filename: '스크린샷 2024-01-09 오후 6.12.25.png',
        metadata: null,
        uploaded: '2024-03-05T08:33:59.794Z',
        requireSignedURLs: false,
        variants: [
          'https://imagedelivery.net/MXlZJaCYonU_kO5E66JLvw/5ccbd20a-49de-48dc-f878-75ece31e7600/public',
          'https://imagedelivery.net/MXlZJaCYonU_kO5E66JLvw/5ccbd20a-49de-48dc-f878-75ece31e7600/adminPost',
          'https://imagedelivery.net/MXlZJaCYonU_kO5E66JLvw/5ccbd20a-49de-48dc-f878-75ece31e7600/origin',
        ],
      },
    },
    {
      id: 37,
      createAt: '2024-03-05T08:34:20.205Z',
      updateAt: '2024-03-05T08:34:20.205Z',
      version: 1,
      title: '8',
      drawingDate: null,
      artType: 'WATERCOLOR',
      canvasSize: '',
      price: 0,
      isSold: false,
      frameType: '',
      contents: '',
      tags: [],
      img: {
        uid: 37,
        id: 'f2cedb2c-0d6e-4797-110d-75d069f9f600',
        filename: '18d4b7abc913b0a.jpg',
        metadata: null,
        uploaded: '2024-03-05T08:34:20.075Z',
        requireSignedURLs: false,
        variants: [
          'https://imagedelivery.net/MXlZJaCYonU_kO5E66JLvw/f2cedb2c-0d6e-4797-110d-75d069f9f600/public',
          'https://imagedelivery.net/MXlZJaCYonU_kO5E66JLvw/f2cedb2c-0d6e-4797-110d-75d069f9f600/origin',
          'https://imagedelivery.net/MXlZJaCYonU_kO5E66JLvw/f2cedb2c-0d6e-4797-110d-75d069f9f600/adminPost',
        ],
      },
    },
    {
      id: 26,
      createAt: '2024-02-21T11:33:42.041Z',
      updateAt: '2024-03-12T06:55:02.272Z',
      version: 5,
      title: '샘플 게시글22',
      drawingDate: null,
      artType: 'ACRYLIC_PAINTING',
      canvasSize: '61cmX65cm',
      price: 1000000,
      isSold: true,
      frameType: '캔버스',
      contents: '내용내용',
      tags: [],
      img: {
        uid: 26,
        id: '8a596c47-4b93-498b-4e8f-50095b767500',
        filename: 'KakaoTalk_20240220_115102262-crop.jpg',
        metadata: null,
        uploaded: '2024-02-21T11:33:41.835Z',
        requireSignedURLs: false,
        variants: [
          'https://imagedelivery.net/MXlZJaCYonU_kO5E66JLvw/8a596c47-4b93-498b-4e8f-50095b767500/adminPost',
          'https://imagedelivery.net/MXlZJaCYonU_kO5E66JLvw/8a596c47-4b93-498b-4e8f-50095b767500/origin',
          'https://imagedelivery.net/MXlZJaCYonU_kO5E66JLvw/8a596c47-4b93-498b-4e8f-50095b767500/public',
        ],
      },
    },
  ],
  meta: {
    totalItems: 12,
    itemCount: 12,
    itemsPerPage: 100,
    totalPages: 1,
    currentPage: 1,
  },
  links: {
    first: 'http://localhost:3000/posts?limit=100',
    previous: '',
    next: '',
    last: 'http://localhost:3000/posts?page=1&limit=100',
  },
};

const postsHandler = [
  http.get(`${BASE_URL}/posts`, () => HttpResponse.json({ ...paginatePosts })),
];

export default postsHandler;
