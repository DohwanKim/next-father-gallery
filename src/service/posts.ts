import queryString from 'query-string';

import { Paginate, PaginatePostsOptions } from '@/types/paginate.type';
import { Post, PostCore } from '@/types/posts.type';
import apiFetch from '@/utils/fetchIntance';

export const getPaginatePosts = async (
  options: PaginatePostsOptions,
): Promise<Paginate<Post>> => {
  const query = queryString.stringify(options, {
    arrayFormat: 'comma',
    skipNull: true,
    skipEmptyString: true,
  });

  return await apiFetch(`/api/posts${query ? `?${query}` : ''}`).then(
    (res) => res.body as Paginate<Post>,
  );
};

export const getPost = async (id: number): Promise<Post> => {
  return await apiFetch(`/api/posts/${id}`).then((res) => res.body as Post);
};

export const getRandomPost = async (id: number): Promise<Post[]> => {
  return await apiFetch(`/api/posts/random-post`)
    .then((res) => res.body as Post[])
    .then((res) => res.filter((post) => post.id !== id).slice(0, 3));
};

export const createPost = async (data: PostCore): Promise<boolean> => {
  return await apiFetch(`/api/posts`, {
    method: 'POST',
    body: data,
  }).then((res) => res.body as boolean);
};

export const updatePost = async (
  id: number,
  data: PostCore,
): Promise<boolean> => {
  return await apiFetch(`/api/posts/${id}`, {
    method: 'PATCH',
    body: data,
  }).then((res) => res.body as boolean);
};

export const deletePost = async (id: number | number[]): Promise<boolean> => {
  const data = Array.isArray(id) ? id : [id];

  return await apiFetch(`/api/posts/${data}`, {
    method: 'DELETE',
  }).then((res) => res.body as boolean);
};
