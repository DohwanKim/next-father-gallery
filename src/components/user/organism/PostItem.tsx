'use client';
import Image from 'next/image';
import Link from 'next/link';
import { AnchorHTMLAttributes, useState } from 'react';

import { Skeleton } from '@/components/ui/skeleton';
import { ImagesVariants } from '@/constants/images.enum';
import { cn } from '@/lib/utils';
import { Post } from '@/types/posts.type';
import { artTypeToKorean, getCFUrl } from '@/utils/common';

interface Props extends Omit<AnchorHTMLAttributes<HTMLAnchorElement>, 'href'> {
  postItem: Post;
}

export const PostItemSkeleton = () => {
  return Array.from({ length: 9 }, (_, index) => (
    <div key={index}>
      <Skeleton className="w-full aspect-square drop-shadow mb-5" />
      <div className={'flex flex-col items-center'}>
        <Skeleton className="h-4 w-[200px] mb-2" />
        <Skeleton className="h-4 w-[100px]" />
      </div>
    </div>
  ));
};

const PostItem = ({ postItem, className, ...props }: Props) => {
  const [originImgLoaded, setOriginImgLoaded] = useState<boolean>(false);
  const { img, title, artType } = postItem;

  return (
    <Link
      href={`/gallery/${postItem.id}`}
      {...props}
      className={cn(
        '[&:hover_img]:scale-105 [&:hover_div_div]:opacity-15',
        className,
      )}
    >
      {img && (
        <div
          className={
            'relative w-full aspect-square mb-2 overflow-hidden drop-shadow'
          }
        >
          {!originImgLoaded && (
            <Image
              fill
              src={getCFUrl(img.id, ImagesVariants.USER_POST_BLUR)}
              alt={''}
              unoptimized
              priority
              className={'object-cover duration-300'}
            />
          )}
          <Image
            fill
            src={getCFUrl(img.id, ImagesVariants.USER_POST)}
            alt={''}
            unoptimized
            className={'object-cover transition-transform duration-300'}
            onLoad={() => setOriginImgLoaded(true)}
          />
          <div
            className={
              'absolute w-full h-full transition-opacity duration-300 opacity-0 bg-black'
            }
          />
        </div>
      )}
      <div className={'text-center text-sm'}>
        <h2>{title}</h2>
        <p className={'font-light'}>{artTypeToKorean(artType)}</p>
      </div>
    </Link>
  );
};

export default PostItem;
