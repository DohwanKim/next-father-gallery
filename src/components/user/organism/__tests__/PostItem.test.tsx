import { fireEvent, render, screen, waitFor } from '@testing-library/react';

import { getPaginatePosts } from '@/service/posts';
import { Paginate } from '@/types/paginate.type';
import { Post } from '@/types/posts.type';

import PostItem from '../PostItem';

describe('PostItem', () => {
  let paginatePosts: Paginate<Post>;

  beforeAll(async () => {
    // MSW DATA
    paginatePosts = await getPaginatePosts({ page: 1, limit: 10 });
  });

  it('데이터를 받아서 랜더한다', async () => {
    const item = paginatePosts.items[0] as Post;
    const { findAllByTitle } = render(<PostItem postItem={item} />);

    expect(findAllByTitle('샘플 게시글1')).toBeTruthy();
  });

  it('a 태그에 올바른 디테일 페이지 url이 있다', async () => {
    const item = paginatePosts.items[0] as Post;
    const { findByRole } = render(<PostItem postItem={item} />);
    const link = await findByRole('link');

    expect(link).toHaveAttribute('href', `/gallery/${item.id}`);
  });

  it('이미지 데이터가 없으면 이미지 태그가 없다', async () => {
    const noImgItem = { ...paginatePosts.items[0], img: null } as Post;
    render(<PostItem postItem={noImgItem} />);
    const img = document.querySelectorAll('img');

    expect(img.length).toBe(0);
  });

  it('이미지 데이터가 있으면 메인 이미지가 로드 후 블러이미지가 감춰지고 메인이미지가 보인다', async () => {
    const item = paginatePosts.items[0] as Post;
    render(<PostItem postItem={item} />);
    const blurImg = screen.getByAltText(`${item.title} blur image`);
    const mainImg = screen.getByAltText(`${item.title} image`);

    expect(blurImg).toHaveClass('opacity-100');
    expect(mainImg).toHaveClass('opacity-0');

    fireEvent.load(mainImg);
    await waitFor(() => {
      expect(blurImg).toHaveClass('opacity-0');
      expect(mainImg).toHaveClass('opacity-100');
    });
  });
});
