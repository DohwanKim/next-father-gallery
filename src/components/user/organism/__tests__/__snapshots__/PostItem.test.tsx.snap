// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostItem matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="[&:hover_img]:scale-105 [&:hover_div_div]:opacity-15"
    href="/gallery/25"
  >
    <div
      class="relative w-full aspect-square mb-2 overflow-hidden"
    >
      <img
        alt="샘플 게시글1 blur image"
        class="object-cover transition-opacity duration-500 opacity-100"
        data-nimg="fill"
        decoding="async"
        fetchpriority="high"
        sizes="100vw"
        src="/_next/image?url=https%3A%2F%2Fimagedelivery.net%2FMXlZJaCYonU_kO5E66JLvw%2Ff0a8866e-2b9a-4329-ba01-0ae7fc210b00%2FuserPostBlur&w=3840&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fimagedelivery.net%2FMXlZJaCYonU_kO5E66JLvw%2Ff0a8866e-2b9a-4329-ba01-0ae7fc210b00%2FuserPostBlur&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fimagedelivery.net%2FMXlZJaCYonU_kO5E66JLvw%2Ff0a8866e-2b9a-4329-ba01-0ae7fc210b00%2FuserPostBlur&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fimagedelivery.net%2FMXlZJaCYonU_kO5E66JLvw%2Ff0a8866e-2b9a-4329-ba01-0ae7fc210b00%2FuserPostBlur&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fimagedelivery.net%2FMXlZJaCYonU_kO5E66JLvw%2Ff0a8866e-2b9a-4329-ba01-0ae7fc210b00%2FuserPostBlur&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fimagedelivery.net%2FMXlZJaCYonU_kO5E66JLvw%2Ff0a8866e-2b9a-4329-ba01-0ae7fc210b00%2FuserPostBlur&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fimagedelivery.net%2FMXlZJaCYonU_kO5E66JLvw%2Ff0a8866e-2b9a-4329-ba01-0ae7fc210b00%2FuserPostBlur&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fimagedelivery.net%2FMXlZJaCYonU_kO5E66JLvw%2Ff0a8866e-2b9a-4329-ba01-0ae7fc210b00%2FuserPostBlur&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fimagedelivery.net%2FMXlZJaCYonU_kO5E66JLvw%2Ff0a8866e-2b9a-4329-ba01-0ae7fc210b00%2FuserPostBlur&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
      <img
        alt="샘플 게시글1 image"
        class="object-cover transition-all duration-300 opacity-0"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        src="https://imagedelivery.net/MXlZJaCYonU_kO5E66JLvw/f0a8866e-2b9a-4329-ba01-0ae7fc210b00/userPost"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
      <div
        class="absolute w-full h-full transition-opacity duration-300 opacity-0 bg-black"
      />
    </div>
    <div
      class="text-center text-sm"
    >
      <h2>
        샘플 게시글1
      </h2>
      <p
        class="font-light"
      >
        수채화
      </p>
    </div>
  </a>
</DocumentFragment>
`;
